flair/data.py

```diff
@@ -830,39 +843,33 @@ def clear_embeddings(self, embedding_names: List[str] = None):
             token.clear_embeddings(embedding_names)
 
     @lru_cache(maxsize=1)  # cache last context, as training repeats calls
-    def left_context(self, context_length: int, respect_document_boundaries: bool = True):
+    def left_context(self, context_length: int, respect_document_boundaries: bool = True) -> List[Token]:
         sentence = self
-        left_context: List[str] = []
-        while True:
+        left_context: List[Token] = []
+        while len(left_context) < context_length:
             sentence = sentence.previous_sentence()
             if sentence is None:
                 break
 
             if respect_document_boundaries and sentence.is_document_boundary:
                 break
 
-            left_context = [t.text for t in sentence.tokens] + left_context
-            if len(left_context) > context_length:
-                left_context = left_context[-context_length:]
-                break
-        return left_context
+            left_context = sentence.tokens + left_context
```

Comment:
`left_context += sentence.tokens` to be consistent with right context

Reply:
Notice that addition on lists is not cummutative, hence your suggestion would lead to wrong results.